src/app/clusters/mode-base-server/mode-base-server.h

```diff
@@ -149,19 +112,56 @@ class Instance : public CommandHandlerInterface, public AttributeAccessInterface
      */
     bool IsSupportedMode(uint8_t mode);
 
-    /**
-     * Creates a mode base cluster instance. The Init() function needs to be called for this instance to be registered and
-     * called by the interaction model at the appropriate times.
-     * @param aEndpointId The endpoint on which this cluster exists. This must match the zap configuration.
-     * @param aClusterId The ID of the ModeBase aliased cluster to be instantiated.
-     * @param aFeature The bitmask value that identifies which features are supported by this instance.
-     */
-    Instance(Delegate * aDelegate, EndpointId aEndpointId, ClusterId aClusterId, uint32_t aFeature);
+private:
+    Delegate * mDelegate;
```

Comment:
if you are initializing mEndpointId and mClusterId, how about setting this to nullptr and setting default values for current mode and feature?

Reply:
I think the only one that makes sense to init here is the CurrentMode as this is the only one we don't get during the instantiation. All the others are passed in the constructor. This would make the constructor code cleaner as the comment against the CurrentMode is moved here. See what you think with this change.